packages/flutter/lib/src/painting/gradient.dart

```diff
@@ -374,16 +374,34 @@ class LinearGradient extends Gradient {
       return b.scale(t);
     if (b == null)
       return a.scale(1.0 - t);
-    final _ColorsAndStops interpolated = _interpolateColorsAndStops(a.colors, a.stops, b.colors, b.stops, t);
```

Comment:
Any reason you can't add the new logic to the `_interpolateColorsAndStops`, so the other Gradient classes can also profit from this behavior?

Reply:
(There's a TODO in that function for exactly this functionality)

Reply:
I've moved the logic there